src/contracts/middleware/StakeRegistry.sol

```diff
@@ -0,0 +1,508 @@
+    );
+
+    constructor(
+        IStrategyManager _strategyManager,
+        IServiceManager _serviceManager
+    ) StakeRegistryStorage(_strategyManager, _serviceManager)
+    // solhint-disable-next-line no-empty-blocks
+    {
+    }
+
+    /**
+     * @notice Adds empty first entries to the dynamic arrays `totalStakeHistory` and `totalOperatorsHistory`,
+     * to record an initial condition of zero operators with zero total stake.
+     * Adds `_quorumStrategiesConsideredAndMultipliers` for each quorum the Registry is being initialized with
+     */
+    function initialize(
+        uint96[] memory _minimumStakeForQuorum,
+        StrategyAndWeightingMultiplier[][] memory _quorumStrategiesConsideredAndMultipliers
+    ) external virtual initializer {
+        _initialize(_minimumStakeForQuorum, _quorumStrategiesConsideredAndMultipliers);
+    }
+
+    function _initialize(
+        uint96[] memory _minimumStakeForQuorum,
+        StrategyAndWeightingMultiplier[][] memory _quorumStrategiesConsideredAndMultipliers
+    ) internal virtual onlyInitializing {
+        // sanity check lengths
+        require(_minimumStakeForQuorum.length == _quorumStrategiesConsideredAndMultipliers.length, "Registry._initialize: minimumStakeForQuorum length mismatch");
+
+        // add the strategies considered and multipliers for each quorum
+        for (uint8 quorumNumber = 0; quorumNumber < _quorumStrategiesConsideredAndMultipliers.length;) {
+            minimumStakeForQuorum[quorumNumber] = _minimumStakeForQuorum[quorumNumber];
+            _createQuorum(_quorumStrategiesConsideredAndMultipliers[quorumNumber]);
+            unchecked {
+                ++quorumNumber;
+            }
+        }
+    }
+
+    /**
+     * @notice Returns the `index`-th entry in the `operatorIdToStakeHistory[operatorId][quorumNumber]` array.
+     * @param quorumNumber The quorum number to get the stake for.
+     * @param operatorId The id of the operator of interest.
+     * @param index Array index for lookup, within the dynamic array `operatorIdToStakeHistory[operatorId][quorumNumber]`.
+     * @dev Function will revert if `index` is out-of-bounds.
+     */
+    function getStakeUpdateForQuorumFromOperatorIdAndIndex(uint8 quorumNumber, bytes32 operatorId, uint256 index)
+        external
+        view
+        returns (OperatorStakeUpdate memory)
+    {
+        return operatorIdToStakeHistory[operatorId][quorumNumber][index];
+    }
+
+    /**
+     * @notice Returns the `index`-th entry in the dynamic array of total stake, `totalStakeHistory` for quorum `quorumNumber`.
+     * @param quorumNumber The quorum number to get the stake for.
+     * @param index Array index for lookup, within the dynamic array `totalStakeHistory[quorumNumber]`.
+     */
+    function getTotalStakeUpdateForQuorumFromIndex(uint8 quorumNumber, uint256 index) external view returns (OperatorStakeUpdate memory) {
+        return totalStakeHistory[quorumNumber][index];
+    }
```

Comment:
is there a reason this function isn't just named `totalStakeHistory` with the mapping being internal (and prepended with an underscore)? seems like a redundant getter function that exists because we don't want the getter function that returns a tuple instead of a struct.

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/a5ec4e86e040c1ff5050b486ac2362b0d3224455